### ISSUE TITLE
jsx/prop-types doesn't trigger with React.forwardRef

### ISSUE THREAD

**Issue by yhaskell** [yannickcr/eslint-plugin-react#1841]
eslint-plugin-react version: `7.9.1`

If we create the functional component, the rule triggers:

```js
const Label = ({ text }) => { 
  return <div>{text}</div>; // 'text' is missing in props validation
}
```

But if we convert this to a React.forwardRef component, it stops triggering:

```js
const Label = React.forwardRef(({ text }, ref) => {
  return <div ref={ref}>{text}</div>; // no error
});
```

**Comment by ljharb** [comment]
Our component detection will need to add support for forwardRef.

**Comment by DavitVosk** [comment]
Any success?
